contrib/python/test/test_images.py

```diff
@@ -82,7 +82,9 @@ def test_get(self):
 
     def test_history(self):
         for count, record in enumerate(self.alpine_image.history()):
-            self.assertEqual(record.id, self.alpine_image.id)
+            expected = record.id == self.alpine_image.id \
+                or record.id == '<missing>'
+            self.assertTrue(expected)
```

Comment:
Any way to verify that we got at least one image id, preferably the right one, and not just a bunch of '<missing>' labels?

Reply:
Hmm, I can get more aggressive with the knowledge we could break again.

Reply:
This is also going to give us less information when `record.id` is a string that differs from both of the allowed values.  How about using [`AssertIn`][1]?

```python
self.AssertIn(record.id, [self.alpine_image.id, '<missing>'])
```

[1]: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertIn

Reply:
> Hmm, I can get more aggressive with the knowledge we could break again.

I prefer the occasional false positive to the occasional false negative, so I'm in favor of @TomSweeneyRedHat's suggestion.  We can always patch later if the stricter test breaks and we decide that we're ok with the broken value.

Reply:
@wking @TomSweeneyRedHat Latest patch has a Counter() in it to determine we're getting the expected values and the correct number of them.   @wking I had forgotten assertIn, thanks for the reminder.  Code looks cleaner now.